internal/serve/auth/signature_verifier.go

```diff
@@ -0,0 +1,157 @@
+package auth
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/strkey"
+	"github.com/stellar/go/support/log"
+)
+
+type SignatureVerifier interface {
+	VerifySignature(ctx context.Context, signatureHeaderContent string, rawReqBody []byte) error
+}
+
+var (
+	ErrStellarSignatureNotVerified = errors.New("neither Signature nor X-Stellar-Signature header could be verified")
+)
+
+type ErrInvalidTimestampFormat struct {
+	TimestampString string
+}
+
+func (e ErrInvalidTimestampFormat) Error() string {
+	return fmt.Sprintf("signature format different than expected. expected unix seconds, got: %s", e.TimestampString)
+}
+
+type ErrExpiredSignatureTimestamp struct {
+	ExpiredSignatureTimestamp time.Time
+	CheckTime                 time.Time
+}
+
+func (e ErrExpiredSignatureTimestamp) Error() string {
+	return fmt.Sprintf("signature timestamp has expired. sig timestamp: %s, check time %s", e.ExpiredSignatureTimestamp.Format(time.RFC3339), e.CheckTime.Format(time.RFC3339))
+}
+
+type StellarSignatureVerifier struct {
+	ServerHostname   string
+	WalletSigningKey string
+}
+
+var _ SignatureVerifier = (*StellarSignatureVerifier)(nil)
+
+// VerifySignature verifies the Signature or X-Stellar-Signature content and checks if the signature is signed for a known caller.
+func (sv *StellarSignatureVerifier) VerifySignature(ctx context.Context, signatureHeaderContent string, rawReqBody []byte) error {
+	t, s, err := ExtractTimestampedSignature(signatureHeaderContent)
+	if err != nil {
+		log.Ctx(ctx).Error(err)
+		var errTimestampFormat *ErrInvalidTimestampFormat
+		if errors.As(err, &errTimestampFormat) {
+			return err
+		}
+		return ErrStellarSignatureNotVerified
+	}
+
+	// 2 minutes in seconds
+	err = VerifyGracePeriodSeconds(t, 2*time.Second)
```

Comment:
The comment says 2 minutes but it being set as 2s. Which of the two do we want to set it to?